### PR TITLE
[M]Switch "big" IntegerField() to DecimalField in core models

### PR DESCRIPTION
**Pull request by Jakub89**
Resolves: #1023

### CODE REVIEW

File: concent_api/core/models.py [hunk 1]
```diff
@@ -637,14 +637,14 @@ def clean(self) -> None:
                 'provider_eth_account': 'Provider ethereum account address must be diffrent than task owner key'
             })
 
-        if not isinstance(self.amount_pending, int) or self.amount_pending < 0:
+        if self.amount_pending < 0:
```
Comment (rwrzesien) [inline concent_api/core/models.py]: I know that this is not related with that changes but as we are here, what about condition when it is equal 0? Neither the condition nor error message cover that case (so on of those should be fixed).
Apart from that, I think it should work still if you check that value type is either `int` or `Decimal`.

File: concent_api/core/models.py [hunk 2]
```diff
@@ -637,14 +637,14 @@ def clean(self) -> None:
...
             raise ValidationError({
-                'amount_pending': 'Amount pending must be an integer and bigger than 0'
+                'amount_pending': 'Amount pending must be bigger than 0'
             })
 
-        if not isinstance(self.amount_paid, int) or self.amount_paid < 0:
+        if self.amount_paid < 0:
```
Comment (rwrzesien) [inline concent_api/core/models.py]: Same here about the type, the message here is ok.

File: concent_api/core/models.py [hunk 3]
```diff
@@ -765,9 +765,9 @@ def clean(self) -> None:
                                           f'(requestor, provider or Concent) must be string and must be exactly '
                                           f'{ETHEREUM_ADDRESS_LENGTH} characters long.'
             })
-        if not isinstance(self.amount, int) or self.amount <= 0:
```
Comment (rwrzesien) [inline concent_api/core/models.py]: Same here about the type, the message is ok here.
Btw. please just add dots at the end of the sentences, I see that sometimes they are and sometimes not, lets keep it consistent (and correct). Could be separate commit.

File: concent_api/core/tests/test_unit_model.py [hunk 4]
```diff
@@ -308,11 +308,6 @@ def test_that_exception_is_raised_when_amount_is_less_then_zero(self):
             self.deposit_claim.clean()
         self.assertIn('amount', exception_info.value.error_dict)
 
-    def test_that_exception_is_raised_when_amount_is_of_wrong_type(self):
-        self.deposit_claim.amount = 5.0
```
Comment (rwrzesien) [inline concent_api/core/tests/test_unit_model.py]: I would suggest keeping this test but changing the value to a non accepted type. This way we still test one branch of the `if` condition (after doing the comments I have added). @dybi do you have opinion on that? Is it better to remove it?

File: concent_api/core/models.py [hunk 5]
```diff
@@ -739,7 +739,7 @@ class DepositClaim(Model):
     payer_deposit_account = ForeignKey(DepositAccount)
     payee_ethereum_address = CharField(max_length=ETHEREUM_ADDRESS_LENGTH)
     concent_use_case = IntegerField()
-    amount = IntegerField()
+    amount = DecimalField(max_digits=78, decimal_places=0)
```
Comment (kbeker) [inline concent_api/core/models.py]: We should have tests for all this model changes. Please add tests which will test maximum possible value for this changed fields

File: concent_api/core/models.py (hunk 1 above)
Comment (dybi) [reply concent_api/core/models.py]: I think that `amount_pending` can be `0` - if all claims have been satisfied.

File: concent_api/core/tests/test_unit_model.py (hunk 4 above)
Comment (dybi) [reply concent_api/core/tests/test_unit_model.py]: If `DecimalField` validations check that for us automatically, then we don't have to add tests for that. If not, then it would be good to have one.

Comment (Jakub89) [reply concent_api/core/tests/test_unit_model.py]: Done

File: concent_api/core/models.py (hunk 5 above)
Comment (Jakub89) [reply concent_api/core/models.py]: Done

File: concent_api/core/models.py (hunk 3 above)
Comment (Jakub89) [reply concent_api/core/models.py]: Type not needed, dot added.

File: concent_api/core/models.py (hunk 1 above)
Comment (rwrzesien) [reply concent_api/core/models.py]: Each of those validations is not actually needed to be performed there, if you want to validate if value is grater than 0 you can set `MinValueValidator(0)` directly on the model. I thought we are doing this for custom error messages - if not, what is the reason for keeping this custom validation?

Comment (Jakub89) [reply concent_api/core/models.py]: @cameel said the same thing on Friday, I can change it to django Validator then.

Comment (rwrzesien) [reply concent_api/core/models.py]: Ok, please do that then.

File: concent_api/core/tests/test_unit_model.py [hunk 6]
```diff
@@ -308,10 +308,15 @@ def test_that_exception_is_raised_when_amount_is_less_then_zero(self):
...
+    def test_that_exception_is_raised_when_amount_is_to_large(self):
```
Comment (kbeker) [inline concent_api/core/tests/test_unit_model.py]: `is_to_large` -> `is_too_large`

Comment (Jakub89) [reply concent_api/core/tests/test_unit_model.py]: Done

File: concent_api/core/models.py (hunk 1 above)
Comment (Jakub89) [reply concent_api/core/models.py]: Done
